scripts/parallel_deblur.py

```diff
@@ -0,0 +1,147 @@
+              show_default=True, help="Sequence trim length")
+@click.option('--min-size', required=False, type=int, default=2,
+              show_default=True, help="Discard sequences with an abundance "
+              "value smaller than min-size")
+@click.option('--negate', '-n', required=False, default=False,
+              show_default=True, type=bool,
+              help="Discard all sequences aligning to the database "
+                   "passed via --ref-fp option")
+@click.option('--threads', '-a', required=False, type=int,
+              default=1, show_default=True,
+              help="Number of threads to use for SortMeRNA")
+@click.option('--delim', required=False, type=str, default='_',
+              show_default=True, help="Delimiter in FASTA labels to separate "
+                                      "sample ID from sequence ID")
+@click.option('--buffer-size', required=False, type=int, default=500,
+              show_default=True, help="The number of sequences to read into "
+              "memory before writing to file (for splitting input file by "
+              "sample)")
+@click.option('--jobs-to-start', '-O', required=False, type=int, default=1,
+              show_default=True,
+              help="Number of jobs to start (if to run in parallel)")
+@click.option('--retain-temp-files', required=False, type=bool, default=False,
+              show_default=True,
+              help="Retain temporary files after parallel runs complete "
+                   "(useful for debugging)")
+@click.option('--suppress-polling', required=False, type=bool, default=False,
+              show_default=True,
+              help="Suppress polling of jobs and merging of results upon "
+                   "completion")
+def workflow(seqs_fp, output_fp, file_type, read_error, mean_error,
+             error_dist, indel_prob, indel_max, trim_length, min_size,
+             ref_fp, ref_db_fp, negate, threads, delim, buffer_size,
+             jobs_to_start, retain_temp_files, suppress_polling):
+    """Launch deblur workflow in parallel"""
+    # If the user provided an error_dist value, we map it to a list of floats
+    if error_dist:
+        error_dist = list(map(float, error_dist.split(',')))
+    out_dir = dirname(output_fp)
+    # Split demultiplexed FASTA on sample IDs
+    out_dir_split = join(out_dir, "split")
+    if not exists(out_dir_split):
+        with open(seqs_fp, 'U') as seqs_f:
+            split_sequence_file_on_sample_ids_to_files(
+                seqs_f,
+                file_type,
+                out_dir_split,
+                buffer_size)
+    # Run deblur in parallel
+    params = {}
+    input_fps = glob('%s/*' % out_dir_split)
+    params['ref_db_fp'] = ref_db_fp
+    params['ref_fp'] = ref_fp
+    params['jobs_to_start'] = jobs_to_start
+    # Number of samples written
+    samples_fp = [f for f in listdir(out_dir_split) if isfile(f)]
+    parallel_runner = ParallelDeblur(
```

Comment:
@ekopylova, is that necessary or is it necessary to just be able to run multiple threads? If the design intent is to go beyond, say, 16 cores then using ipython parallel makes sense